docs/development/typescript.md

```diff
@@ -0,0 +1,64 @@
+# Typescript
+
+**Question**: should we recommend using typescript over javascript?
+
+**TODO:** advantages of typescript over javascript?
+
+Node.js does not support running typescript natively, so it must first be transpiled to JavaScript. There are 2 widely used transpilers:
+
+1. `tsc` - the compiler shipped with [typescript](https://www.npmjs.com/package/typescript), supports type checking and transpiles to JavaScript
+
+2. [`babel`](https://babeljs.io) - can transpile to javascript using the [`@babel/preset-typescript`](https://babeljs.io/docs/en/babel-preset-typescript) preset, does _not_ support type checking
+
+There are other transpilers such as [esbuild](https://esbuild.github.io) and [swc](https://swc.rs) but these are not yet widely adopted and not currently recommended.
+
+Avoid mixing transpilers on the same project as this can lead to inconsistencies. One common mistake is using `tsc` for code transpilation but `babel` (the [default](https://jestjs.io/docs/getting-started#using-typescript)) for jest tests.
+
+More information on choosing a transpiler can be found in the [typescript documentation](https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html).
+
+## Transpiling in development
+
+TODO: talk about nodemon with babel-node, `tsc -w`, `ts-node`
+
+Many code editors (such as Visual Studio Code) provide support for type checking of typescript. If you're using one of these editors, you can speed up transpilation by using babel without type checking. **TODO**: check if there's a way to turn this on for `tsc` as well.
```

Comment:
This makes me think we might want to cover one of the advantages of getting better contextual help in a code editor somewhere?